ios-app/UI/DRMKeySessionDelegate.swift

```diff
@@ -0,0 +1,112 @@
+import Foundation
+import AVKit
+import Alamofire
+
+
+class DRMKeySessionDelegate: NSObject, AVContentKeySessionDelegate {
+    let drmLicenseURL: String?
+    let instituteSettings = DBManager<InstituteSettings>().getResultsFromDB()[0]
+    
+    public required init?(drmLicenseURL: String?) {
+        self.drmLicenseURL = drmLicenseURL
+    }
+    
+    @available(iOS 10.3, *)
+    func contentKeySession(_ session: AVContentKeySession, didProvide keyRequest: AVContentKeyRequest) {
+        
+        guard let contentId = keyRequest.identifier as? String else {
+            keyRequest.processContentKeyResponseError(DRMError.noContentId)
+            return
+        }
+        
+        fetchDRMKey(keyRequest, contentId)
+    }
+    
+    
+    func fetchDRMKey(_ keyRequest: AVContentKeyRequest, _ contentId: String) {
+        let contentIdData = contentId.data(using: String.Encoding.utf8)
+
+        do {
+            let certificateData = try getFairplayCertificateData()
+            keyRequest.makeStreamingContentKeyRequestData(forApp: certificateData, contentIdentifier: contentIdData, options: [AVAssetResourceLoadingRequestStreamingContentKeyRequestRequiresPersistentKey: true as AnyObject]) { spcData, spcError in
+                
+                guard let spcData = spcData else {
+                    let error = spcError ?? DRMError.noSPCData
+                    keyRequest.processContentKeyResponseError(error)
+                    return
+                }
+                
+                self.requestContentKeyFromKeySecurityModule(spcData: spcData, assetID: contentId) {
+                    data in
+                    
+                    let keyResponse = AVContentKeyResponse(fairPlayStreamingKeyResponseData: data!)
+                    keyRequest.processContentKeyResponse(keyResponse)
+                }
+                
+            }
+            
+        } catch {
+            return keyRequest.processContentKeyResponseError(DRMError.licenseError)
+        }
+    }
+    
+    func getFairplayCertificateData() throws -> Data {
+        let certUrl = URL(string: instituteSettings.fairplayCertificateUrl)
+        var applicationCertificate: Data? = nil
+        do {
+            applicationCertificate = try Data(contentsOf:certUrl!)
+        } catch {
+            throw DRMError.certificateError
+        }
+        
+        return applicationCertificate!
+    }
+    
+    func getLicenseURL(completion: @escaping(String?, TPError?) -> Void) {
```

Comment:
`getLicenseURL` should be after `requestContentKeyFromKeySecurityModule`

![image](https://user-images.githubusercontent.com/328112/143383024-69223506-e250-4c43-91be-9253dba6f3c6.png)
![image](https://user-images.githubusercontent.com/328112/143383096-6d4fac43-2780-4f8d-beb3-d80c50a25f4c.png)